### PR TITLE
Fix suback packet.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/packets.toit [hunk 1]
```diff
@@ -229,17 +229,20 @@ class SubscribePacket extends Packet:
...
 
-  constructor.deserialize reader/reader.BufferedReader:
+  constructor.deserialize reader/reader.BufferedReader size/int:
     data := reader.read_bytes 2
     packet_id = binary.BIG_ENDIAN.uint16 data 0
-    qos = reader.read_byte
-    if qos == 127: throw "QoS negotiation failed"
+    consumed := 0
+    qos = []
+    while consumed < size:
```
Comment (kasperl) [inline src/packets.toit:243]: Where is `consumed` supposed to be incremented?

File: src/packets.toit [hunk 2]
```diff
@@ -229,17 +229,20 @@ class SubscribePacket extends Packet:
...
     super TYPE
 
-  constructor.deserialize reader/reader.BufferedReader:
+  constructor.deserialize reader/reader.BufferedReader size/int:
     data := reader.read_bytes 2
     packet_id = binary.BIG_ENDIAN.uint16 data 0
-    qos = reader.read_byte
-    if qos == 127: throw "QoS negotiation failed"
+    consumed := 0
+    qos = []
```
Comment (kasperl) [inline src/packets.toit:242]: Wouldn't it be more logical to allocate this with the right size?

File: src/packets.toit (hunk 1 above)
Comment (floitsch) [reply src/packets.toit]: done.

File: src/packets.toit (hunk 2 above)
Comment (floitsch) [reply src/packets.toit]: done.

File: src/packets.toit [hunk 3]
```diff
@@ -229,17 +229,20 @@ class SubscribePacket extends Packet:
 class SubAckPacket extends Packet implements PacketIDAck:
   static TYPE ::= 9
 
-  packet_id/int
-  qos/int
+  /** The qos value for a failed subscription. */
+  static FAILED_SUBSCRIPTION_QOS ::= 0x80
```
Comment (kasperl) [inline src/packets.toit:233]: So the 127 was just the wrong constant too?

File: examples/subscribe_many.toit [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2021 Toitware ApS.
```
Comment (kasperl) [inline examples/subscribe_many.toit:1]: 2022

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/packets.toit (hunk 3 above)
Comment (floitsch) [reply src/packets.toit:233]: The spec says 0x80...

**Pull request merged**